geode-redis/src/integrationTest/java/org/apache/geode/redis/internal/executor/hash/AbstractHScanIntegrationTest.java

```diff
@@ -60,34 +65,56 @@ public void tearDown() {
     jedis.close();
   }
 
+  /********* Parameter Checks **************/
+
   @Test
-  public void givenNoKeyArgument_returnsWrongNumberOfArgumentsError() {
+  public void givenLessThanTwoArguments_returnsWrongNumberOfArgumentsError() {
     assertThatThrownBy(() -> jedis.sendCommand(Protocol.Command.HSCAN))
         .hasMessageContaining("ERR wrong number of arguments for 'hscan' command");
-  }
 
-  @Test
-  public void givenNoCursorArgument_returnsWrongNumberOfArgumentsError() {
     assertThatThrownBy(() -> jedis.sendCommand(Protocol.Command.HSCAN, "key!"))
         .hasMessageContaining("ERR wrong number of arguments for 'hscan' command");
   }
 
   @Test
-  public void givenArgumentsAreNotOddAndKeyExists_returnsSyntaxError() {
+  public void givenMatchArgumentWithoutPatternOnExistingKey_returnsSyntaxError() {
+    jedis.hset("key", "b", "1");
+
+    assertThatThrownBy(() -> jedis.sendCommand(Protocol.Command.HSCAN, "key", "0", "Match"))
+        .hasMessageContaining(ERROR_SYNTAX);
+  }
+
+  @Test
+
+  public void givenMatchArgumentWithoutPatternOnNonExistentKey_returnsEmptyArray() {
+    jedis.hset("key", "b", "1");
+
+    ScanParams scanParams = new ScanParams();
+    scanParams.match("");
+
+    ScanResult<Map.Entry<String, String>> result =
+        jedis.hscan("key", "0", scanParams);
+
+    assertThat(result.getResult()).isEmpty();
+  }
```

Comment:
The name of this test seems inaccurate.  It says it is a `NonExistentKey` however, we do an hset beforehand with the key, so it should exist with data in the cache. Also, we say we don't provide a pattern, but empty string actually is a pattern.  What we could do (if we want to test a match argument without a pattern on a nonexistent key), is not do the hset, then do:
```
List<Object> result =
        (List<Object>)  jedis.sendCommand(Protocol.Command.HSCAN, "key1", "0", "Match");

assertThat((List<String>) result.get(1)).isEmpty();
```

Reply:
thanks-  will revert to original send command

Reply:
and thanks for catching the name